app/models/concerns/hostext/search.rb

```diff
@@ -216,6 +216,8 @@ def param_conditions(p)
               conditions << "hosts.hostgroup_id IN (#{param.hostgroup.subtree_ids.join(', ')})"
             when 'HostParameter'
               conditions << "hosts.id = #{param.reference_id}"
+            when 'SubnetParameter'
+              conditions << "hosts.subnet_id = #{param.reference_id}"
```

Comment:
@sean797 there is no subnet_id in hosts table so this needs to add a `join`.
I just opened a bug for the same problem with `domain_id`: http://projects.theforeman.org/issues/13937
This can be removed and added with the domain fix, fixed here or wait until 13937 is fixed and then change it.